src/lib/expression/pqp_subquery_expression.cpp

```diff
@@ -43,11 +43,20 @@ std::string PQPSubqueryExpression::as_column_name() const {
 }
 
 bool PQPSubqueryExpression::_shallow_equals(const AbstractExpression& expression) const {
-  Fail("Can't compare PQPSubqueryExpressions");
+  // For deep copies of this expression, the PQP is recreated. As such, the copy is considered unequal to its origin.
```

Comment:
This is a bit cryptic. Prefix comment with "we're just comparing shared pointer addressees here, because it is unfeasable to deeply compare PQPs"?

Reply:
"un-/infeasible" is cryptic itself. Is there a logical reason why one should not do it, a technical one, one that is related to the performance, or was the developer simply lazy? Suggestion:

```
  // For deep copies of this expression, the PQP is recreated. It is not only difficult to correctly compare PQPs, but
  // it also has unclear semantics. is an executed PQP equal to a non-executed one? How about two executed ones?
  // As such, we only report equality if two PQPSubqueryExpression refer to the very same PQP, meaning that a deep copy
  // is not equal to its source. If this ever becomes an issue, it should be easy to spot.
```

Reply:
much bedda